### DISCUSSION TITLE
how to perform fast integration?

### DISCUSSION THREAD

**Discussion by dahun73** [satijalab/seurat#6161] [Q&A]
Hello everyone,

I'm trying to integrate 8 samples(Control 4 and Case 4). Also, each sample has almost 10,000 cells. So, I have tried to integrate total 80,000 cells. When I tried to perform integration, it took a very long time(maybe 3~4 hours?). I know it depends on my server but I want to know other solution. the number of cpu of server desktop is 32 cores and memory is 256Gb. Is there any good idea perform integration faster?

I also upload my integration code below:

```
### Integration ###

inte.sce <- FindIntegrationAnchors(object.list = list(Control_rep1,Control_rep2,Control_rep3,Control_rep4,
                                                      Case_rep1,Case_rep2,Case_rep3,Case_rep4), dims=1:30)
inte.combined <- IntegrateData(anchorset = inte.sce)

DefaultAssay(inte.combined) <- "integrated"

inte.combined <- ScaleData(inte.combined, verbose = FALSE)
inte.combined <- RunPCA(inte.combined, npcs = 30, verbose = FALSE)

inte.combined <- RunUMAP(inte.combined, reduction = "pca", dims = 1:30)
inte.combined <- FindNeighbors(inte.combined, reduction = "pca", dims = 1:30)
inte.combined <- FindClusters(inte.combined, resolution = 0.5)

```

Thank you in advance!!

**Comment by klpara** [comment]
https://github.com/satijalab/seurat/discussions/3999#discussioncomment-326054 

May be helpful to you.
